src/goto-programs/interpreter_evaluate.cpp

```diff
@@ -959,16 +928,43 @@ void interpretert::evaluate(
     }
     else if(!address.is_zero())
     {
+      if(
+        expr.id() == ID_byte_extract_little_endian ||
+        expr.id() == ID_byte_extract_big_endian)
+      {
+        mp_vectort extract_from;
+        evaluate(expr.op0(), extract_from);
+        INVARIANT(
+          !extract_from.empty(),
+          "evaluate_address should have returned address == 0");
+        const mp_integer memory_offset =
+          address - evaluate_address(expr.op0(), true);
+        const typet &target_type = expr.type();
+        mp_integer target_type_leaves;
+        if(
+          !count_type_leaves(target_type, target_type_leaves) &&
+          target_type_leaves > 0)
+        {
+          dest.insert(
+            dest.end(),
+            extract_from.begin() + numeric_cast_v<std::size_t>(memory_offset),
+            extract_from.begin() +
+              numeric_cast_v<std::size_t>(memory_offset + target_type_leaves));
+          return;
+        }
+        // we fail
```

Comment:
This comment doesn't seem quite right - we don't fail - we carry on to the next if - perhaps a throw here, or a return?